### DISCUSSION TITLE
Are chunked HTTP requests supported?

### DISCUSSION THREAD

**Discussion by midn1** [cesanta/mongoose#2998] [Ask questions, get answers] [answered]
Hello,

I see in the examples one for the case of a streaming server. Is the reverse case supported? I would like to accept a video stream through HTTP from programs including FFmpeg, which do so with a chunked transfer encoding.

If yes, is the approach similar, with `MG_EV_READ`? Will that interfere with the fact the server also handles WebSocket connections at the same time?

**Comment by scaprile** [comment, marked as answer]
Please see our [documentation](https://mongoose.ws/documentation/),  and follow the guidelines in our [tutorials](https://mongoose.ws/documentation/#tutorials-and-examples).

I don't see where what you describe is different from an HTTP client downloading a file or an HTTP server being uploaded a file.
We never said in any document that there is any interference among servers.

**Reply by midn1** [reply]
It's different because in one case, it's an HTTP client. In the other, it's an HTTP server.

>  We never said in any document that there is any interference among servers.

Well, as long as you don't say it, I guess can't exist, huh.

Pro-tip: few people will be so enamored by your library they'll gladly waste their time on something they're not sure will even work.

**Reply by scaprile** [reply]
I think you should be fine assuming that I do know the difference between a client and a server, and that if I told you that, it is because I know what I'm saying. It is your choice, of course.
My pro tip is that if you see all the docs and examples, you'll know how to do things, and "does not work" is not in my vocabulary for an open source library written in C, with clear examples and users like NASA. But it is your time, so you are free to do as you like.

WRT chunked transfer encoding specifically, we simplified handling some versions ago. If you do need that, and you happen to have an issue, then open an issue and we will check and find a solution. Our current handling of it is in our current version of the docs and tutorials. Oh, and you can always bypass the HTTP handler and do it yourself, on your own or calling the parser functions, thing that is also explained in the docs and tutorials.
